src/flb_output.c

```diff
@@ -208,7 +208,12 @@ int flb_output_task_flush(struct flb_task *task,
         }
 
         flb_task_users_inc(task);
-        flb_coro_resume(out_coro->coro);
+        ret = flb_pipe_w(config->ch_self_events[1], &out_coro,
+                        sizeof(struct flb_output_coro*));
```

Comment:
Add coroutine to task pipe rather than processing in place and blocking thread dispatches and other prioritized events.